app/src/main/java/de/moyapro/idle/domain/Trait.kt

```diff
@@ -5,44 +5,58 @@ package de.moyapro.idle.domain
 import kotlin.math.pow
 
 sealed class Trait(val level: Int = 1) {
-    abstract fun influence(consumption: Consumption)
+    abstract fun influence(consumption: Consumption): Consumption
 }
 
-class GrowthTrait() : Trait() {
-    override fun influence(consumption: Consumption) = Unit
+/**
+ * Determin which of the available supply is consumeable/reachable by the current species
+ * This may also introcude a factor so that only a fraction of all available resources can be consumed
+ */
+abstract class SupplyModifyingTrait : Trait() {
+    abstract fun influence(supply: Resources): Resources
+}
+
+
+class GrowthTrait : Trait() {
+    override fun influence(consumption: Consumption): Consumption = consumption
     fun influence(growthRate: Double) = growthRate.pow(level + 1)
 }
 
-class EnergySaver() : Trait() {
-    override fun influence(consumption: Consumption) {
-        consumption.times(energyFactor = .9)
+class EnergySaver : Trait() {
+    override fun influence(consumption: Consumption): Consumption {
+        return consumption.times(energyFactor = .9)
     }
 }
 
-class WaterSaver() : Trait() {
-    override fun influence(consumption: Consumption) {
-        consumption.times(waterFactor = .9)
+class WaterSaver : Trait() {
+    override fun influence(consumption: Consumption): Consumption {
+        return consumption.times(waterFactor = .9)
     }
 }
 
-class MineralSaver() : Trait() {
-    override fun influence(consumption: Consumption) {
-        consumption.times(mineralsFactor = .9)
+class MineralSaver : Trait() {
+    override fun influence(consumption: Consumption): Consumption {
+        return consumption.times(mineralsFactor = .9)
     }
 }
 
-class EvolutionBooster() : Trait() {
-    override fun influence(consumption: Consumption) {
-        consumption.times(evolutionPointsFactor = 1.15)
+class EvolutionBooster : Trait() {
+    override fun influence(consumption: Consumption): Consumption {
+        return consumption.times(evolutionPointsFactor = 1.15)
     }
 }
 
-class Predator(private val prey: Species) : Trait() {
-    override fun influence(consumption: Consumption) {
+class Predator(private val prey: Species) : SupplyModifyingTrait() {
+    override fun influence(supply: Resources): Resources {
+        return supply.times(0.0)
```

Comment:
```suggestion
        return supply * 0.0
```

Reply:
Ein Räuber braucht kein Wasser? Hmm... 😁

Reply:
https://www.songtexte.com/songtext/subway-to-sally/julia-und-die-rauber-2bdccc22.html

Räuber saufen Blut!